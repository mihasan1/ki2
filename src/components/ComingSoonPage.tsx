import React from "react";

import { Message, MessageHeader, MessageBody, Content, Icon } from "bloomer";

import Layout from "./../layouts/Layout";

import { PageWithLocation } from "./../types";

const ComingSoonPage: React.FC<PageWithLocation> = ({ location }) => (
	<Layout location={location}>
		<Message isColor="warning">
			<MessageHeader>
				<p>Увага! ВАЖЛИВЕ ПОВІДОМЛЕННЯ!</p>
			</MessageHeader>
			<MessageBody>
				<Content>
					<h3>Ця сторінка на даний момент незаповнена</h3>
					<br />
					<h4>
						Але це не на довго, адже ми вже працюємо над цим{" "}
						<Icon hasTextColor="danger" className="fa fa-heart" />
					</h4>
					<br />
					<a href="/main">Повернутися на головну</a>
				</Content>
			</MessageBody>
		</Message>
	</Layout>
);

export default ComingSoonPage;
