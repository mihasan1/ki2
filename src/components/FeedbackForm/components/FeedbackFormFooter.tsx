import React from "react";

import { ModalCardFooter, Button, Help } from "bloomer";
import { ResponseStatusOK } from "./../store/types";

export interface FeedbackFormFooterProps {
	cancelHandler?: () => void;
	errorStatus?: boolean;
	isLoading?: boolean;
	responseStatusOK?: ResponseStatusOK;
	darkmode?: boolean;
}

export const FeedbackFormFooter: React.FC<FeedbackFormFooterProps> = ({
	cancelHandler = () => {},
	errorStatus = false,
	isLoading = false,
	responseStatusOK = null,
	darkmode = false,
}) => (
	<ModalCardFooter>
		<Button
			form="feedback_form"
			type="submit"
			isColor={errorStatus ? "danger" : "success"}
			isLoading={isLoading}
			disabled={errorStatus || isLoading}
			isOutlined={darkmode}
		>
			Відправити
		</Button>
		<Button isColor="warning" onClick={cancelHandler} isOutlined={darkmode}>
			Відмінити
		</Button>
		{responseStatusOK !== null &&
			(responseStatusOK ? (
				<Help isColor="success">Все успішно відправлено!</Help>
			) : (
				<Help isColor="danger">Щось пішло не так! Спробуйте знову.</Help>
			))}
	</ModalCardFooter>
);
