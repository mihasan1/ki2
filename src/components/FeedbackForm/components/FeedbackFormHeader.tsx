import React from "react";

import { ModalCardTitle, ModalCardHeader, Delete } from "bloomer";

export const FeedbackFormHeader = ({
	closeModal = () => {},
	darkmode = false,
}) => (
	<ModalCardHeader>
		<ModalCardTitle hasTextColor={darkmode ? "primary" : ""}>
			Опишіть свою знахідку
		</ModalCardTitle>
		<Delete onClick={closeModal} isSize="medium" isHidden="mobile" />
	</ModalCardHeader>
);
