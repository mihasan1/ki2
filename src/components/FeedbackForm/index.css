.modal-card {
	--bg-light: rgba(10 10 10 / 5%);
	--bg-dark: rgba(10 10 10 / 15%);

	--text-light: hsl(220, 5%, 15%);
	--text-dark: hsl(235, 50%, 95%);
}

.modal-card .input,
.modal-card .textarea {
	background: var(--bg-light);
	border: none;
	box-shadow: none;
	border-bottom: 1px solid rgb(150, 164, 194);
	border-radius: 0;
	color: var(--text-light);
}

.dark-mode .modal-card header {
	overflow-x: auto;
}

.dark-mode .modal-card button.delete {
	filter: invert(1);
}

.dark-mode .modal-card * {
	background: var(--dark-mode-bg);
}

.dark-mode .modal-card label {
	color: var(--dark-mode-text-color);
}

.dark-mode .modal-card .input::placeholder,
.dark-mode .modal-card .textarea::placeholder {
	color: aliceblue;
}

.dark-mode .modal-card .input,
.dark-mode .modal-card .textarea {
	background: var(--bg-dark);
	color: var(--text-dark);
}
