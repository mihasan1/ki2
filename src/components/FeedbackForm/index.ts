export { default as FeedbackForm } from "./FeedbackForm";
export { FeedbackFormFooter, FeedbackFormHeader } from "./components";
