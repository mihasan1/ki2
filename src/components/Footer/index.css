.dark-mode #footer {
	background-color: var(--dark-mode-bg) !important;
	color: var(--dark-mode-text-color) !important;
}
