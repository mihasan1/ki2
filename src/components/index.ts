export { LinkDropdownGroup } from "./LinkDropdownGroup";
export { RawMarkdown, RawMarkdownPage } from "./RawMD";
export { default as NewsPreview } from "./NewsPreview";

export { Cloud, Airplane } from "./SkyObjects";

export { default as NavbarLink } from "./NavbarLink";
export { default as Header } from "./Header";
export { default as Footer } from "./Footer";

export {
	FeedbackForm,
	FeedbackFormFooter,
	FeedbackFormHeader,
} from "./FeedbackForm";

export { default as ComingSoonPage } from "./ComingSoonPage";
