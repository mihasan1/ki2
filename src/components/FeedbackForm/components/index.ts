import { FeedbackFormHeader } from "./FeedbackFormHeader";
import { FeedbackFormFooter } from "./FeedbackFormFooter";

export { FeedbackFormHeader, FeedbackFormFooter };
