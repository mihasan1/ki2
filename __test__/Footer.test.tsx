import React from "react";
import renderer from "react-test-renderer";

import { Footer } from "./../src/components";

test("renders correctly", () => {
	const tree = renderer.create(<Footer />).toJSON();

	expect(tree).toMatchSnapshot();
});
