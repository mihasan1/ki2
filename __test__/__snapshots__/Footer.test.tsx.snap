// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="footer"
  id="footer"
>
  <div
    className="container"
  >
    <div
      className="content"
    >
      <div
        className="columns"
      >
        <div
          className="column is-half"
        >
          <p>
            Made with
            <span
              className="icon"
            >
              <i
                aria-hidden="true"
                className="has-text-danger fa fa-heart"
              />
            </span>
            by 
            <a
              href="https://github.com/faramozzayw"
            >
              faramozzayw
            </a>
          </p>
        </div>
        <div
          className="column is-half has-text-right"
        >
          <button
            className="button is-warning"
            onClick={[Function]}
            type="button"
          >
            Знайшли помилку?
          </button>
          <div
            className="modal has-text-left"
          >
            <div
              className="modal-background"
              onClick={[Function]}
            />
            <div
              className="modal-card"
            >
              <header
                className="modal-card-head"
              >
                <h1
                  className="modal-card-title"
                >
                  Опишіть свою знахідку
                </h1>
                <button
                  className="delete is-medium is-hidden-mobile"
                  onClick={[Function]}
                  type="button"
                />
              </header>
              <section
                className="modal-card-body"
              >
                <form
                  id="feedback_form"
                  onSubmit={[Function]}
                >
                  <fieldset
                    disabled={false}
                  >
                    <div
                      className="field"
                    >
                      <label
                        className="label"
                      >
                        Короткий опис проблеми
                      </label>
                      <div
                        className="control"
                      >
                        <input
                          className="input"
                          onChange={[Function]}
                          placeholder="Короткий опис знахідки"
                          required={true}
                          type="text"
                          value=""
                        />
                        <p
                          className="help is-success has-text-right"
                        >
                          100
                          /
                          100
                        </p>
                      </div>
                    </div>
                    <div
                      className="field"
                    >
                      <label
                        className="label"
                      >
                        Детальний опис...
                      </label>
                      <div
                        className="control"
                      >
                        <textarea
                          className="textarea"
                          onChange={[Function]}
                          placeholder="Детальний опис..."
                          required={true}
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      className="field"
                    >
                      <div
                        className="control"
                        onChange={[Function]}
                      >
                        <label
                          className="radio"
                        >
                          <input
                            defaultChecked={true}
                            name="req_type"
                            type="radio"
                            value="bug"
                          />
                           
                          Помилка
                           
                        </label>
                        <label
                          className="radio"
                        >
                          <input
                            name="req_type"
                            type="radio"
                            value="proposal"
                          />
                           
                          Побажання
                           
                        </label>
                        <label
                          className="radio"
                        >
                          <input
                            name="req_type"
                            type="radio"
                            value="feature"
                          />
                           
                          Ідея
                           
                        </label>
                      </div>
                    </div>
                  </fieldset>
                </form>
              </section>
              <footer
                className="modal-card-foot"
              >
                <button
                  className="button is-success"
                  disabled={false}
                  form="feedback_form"
                  type="submit"
                >
                  Відправити
                </button>
                <button
                  className="button is-warning"
                  onClick={[Function]}
                  type="button"
                >
                  Відмінити
                </button>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="theme-switch-wrapper"
    >
      <label
        className="theme-switch"
        htmlFor="checkbox"
      >
        <input
          checked={false}
          id="checkbox"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          className="slider round"
        />
      </label>
      <em>
        Увімкнути тьомну тему!
      </em>
    </div>
  </div>
</footer>
`;
